File: frontend/src/interfaces/domain.ts
import { getDomainPluginInstance, getDomainPluginInstances } from "@/services";
import message from "antd/lib/message";
import { WasmPluginData } from "./route";

export interface Domain {
  name: string;
  version?: string;
  certIdentifier?: string;
  enableHttps?: string;
  [propName: string]: any;
}

export const Protocol = {
  http: 'HTTP',
  https: 'HTTPS',
};

export enum EnableHttpsValue {
  on = 'on',
  off = 'off',
  force = 'force',
}

export interface DomainResponse {
  data: Domain[];
  pageNum: number;
  pageSize: number;
  total: number;
}

export const DEFAULT_DOMAIN = "higress-default-domain";

export const fetchPluginsByDomain = async (record: Domain): Promise<WasmPluginData[]> => {
  const data: Record<string, WasmPluginData[]> = {};
  try {
    const response = await getDomainPluginInstances(record.name);
    const plugins = response.map((plugin: { pluginName: any; description: any; enabled: any; internal: any }) => {
      return {
        ...plugin,
        name: plugin.pluginName,
        enabled: plugin.enabled,
        internal: plugin.internal,
      };
    });
    data[record.name] = plugins || [];
  } catch (error) {
    message.error(`Failed to fetch strategies: ${error.message || error}`);
  }

  return data[record.name] || [];
};

File: backend/sdk/src/main/java/com/alibaba/higress/sdk/model/Domain.java
package com.alibaba.higress.sdk.model;

import io.swagger.v3.oas.annotations.media.Schema;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
@Schema(description = "网关域名配置类，用于管理网关的域名信息。")
public class Domain implements VersionedDto {

    public static class EnableHttps {
        public static final String OFF = "off";
        public static final String ON = "on";
        public static final String FORCE = "force";
    }

    @Schema(description = "域名名称")
    private String name;

    @Schema(description = "域名版本。更新时必需。")
    private String version;

    @Schema(description = "HTTPS 配置", allowableValues = {EnableHttps.OFF, EnableHttps.ON, EnableHttps.FORCE})
    private String enableHttps;

    @Schema(description = "证书标识符")
    private String certIdentifier;
}
